### PR TITLE
[DOCS] Document how CCS handles cluster-level settings

### LINKED ISSUE #49864: [DOCS] Add more detail to CCS docs around how ccs_minimize_roundtrips affects shard limits
Request to update [How cross-cluster search works](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html#ccs-works)

Paraphrasing as per @jimczi :  

Limits like `pre_filter_shard_size`, `max_concurrent_shard_requests`, and most importantly `action.search.shard_count.limit` apply locally to each remote cluster when cross-cluster search is running in `ccs_minimize_roundtrips: true` mode, which is the default.     Importantly, these limits are applied globally across the entire scope of the CCS when searching with `ccs_minimize_roundtrips: false`

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the cross-cluster search (CCS) documentation to note how cluster-level settings are applied.

When `ccs_minimize_roundtrips` is `true`, each cluster applies its own cluster-level settings to the request.

When `ccs_minimize_roundtrips` is `false`, cluster-level settings for the local cluster is used. This includes shard limit settings, such as `action.search.shard_count.limit`, `pre_filter_shard_size`, and `max_concurrent_shard_requests`. If these limits are set too low, the request could be rejected.

Closes #49864.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/modules/cross-cluster-search.asciidoc [hunk 1]
```diff
@@ -340,6 +341,15 @@ image:images/ccs/ccs-min-roundtrip-cluster-results.svg[]
 . The coordinating node sends a search request to each shard, including those in
 its own cluster. Each shard performs the search request independently.
 +
+[WARNING]
+====
+Cluster-level settings from the _local_ cluster are applied to each request.
```
Comment (javanna) [inline docs/reference/modules/cross-cluster-search.asciidoc]: this is the section about not minimizing roundtrips right? I think "request" here may be confusing to users. The way I would try to explain this is that when you don't minimize roundtrips, the whole search is executed almost as if all the data was in a single cluster (the coordinating one, meaning that cluster where the coordinating node is located), hence the limits need to account for that. Is that what you meant?

**Review by javanna — COMMENTED** [review]
I left one comment, thanks for your patience ;)

File: docs/reference/modules/cross-cluster-search.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/modules/cross-cluster-search.asciidoc]: Thanks @javanna. That explanation was very helpful!

I've reworded the warning with 0b8c2619162fdb8789e5b40cd655d63501775885.

**Review by javanna — APPROVED** [review]
LGTM thank you!

**Comment by jrodewig** [comment]
Thanks @javanna!

**Comment by jrodewig** [comment]
### Backport commits
master 67f6840 
7.x  43376c6e061
7.6 953e4549b5bb1042cca77d600ad51bb159d3c0d6

**Pull request merged**